greenhouse/diskcache/cache.go

```diff
@@ -190,26 +190,27 @@ func (c *Cache) Delete(key string) error {
 }
 
 // MonitorDiskAndEvict loops monitoring the disk, evicting cache entries
-// when the disk passes either minPercentBlocksFree or minPercentFilesFree
+// when the disk passes either minPercentBlocksFree until the disk is above
+// evictUntilPercentBlocksFree
 func (c *Cache) MonitorDiskAndEvict(
 	interval time.Duration,
-	minPercentBlocksFree, minPercentFilesFree float64,
-	evictUntilPercentBlocksFree, evictUntilPercentFilesFree float64,
+	minPercentBlocksFree, evictUntilPercentBlocksFree float64,
 ) {
 	// forever check if usage is past thresholds and evict
-	for range time.Tick(interval) {
-		blocksFree, filesFree, err := diskutil.GetDiskUsage(c.diskRoot)
+	ticker := time.NewTicker(interval)
+	for ; true; <-ticker.C {
```

Comment:
see: https://github.com/golang/go/issues/17601#issuecomment-311955879, this lets `for time.Tick(interval) { ... }` be transformed to behave more like `do { ... } while time.Tick(interval)`, which is nice for redeploying since we immediately check the disk instead of waiting `<interval>`